crates/cli/src/hostname.rs

```diff
@@ -25,10 +25,8 @@ pub fn hostname() -> io::Result<OsString> {
     }
     #[cfg(not(any(windows, unix)))]
     {
-        io::Error::new(
-            io::ErrorKind::Other,
-            "hostname could not be found on unsupported platform",
-        )
+        // backup solution for systems that do not have gethostname():
+        Ok(OsString::from("localhost"))
```

Comment:
I'm not a huge fan of giving wrong answers like this. And it seems like WASI can't really support hyperlinks anyway, due to the ban on absolute paths. So I think this change should probably be reverted.

Reply:
The problem I was trying to fix here, and that's also a question I have about setting up CI, is that the current version of the code causes a compilation error:    
```
Compiling grep-cli v0.1.10 (/Users/holzschu/src/Xcode_iPad/ripgrep_clone/crates/cli)
error[E0308]: mismatched types
  --> crates/cli/src/hostname.rs:28:9
   |
16 |   pub fn hostname() -> io::Result<OsString> {
   |                        -------------------- expected `Result<OsString, std::io::Error>` because of return type
...
28 | /         io::Error::new(
29 | |             io::ErrorKind::Other,
30 | |             "hostname could not be found on unsupported platform",
31 | |         )
   | |_________^ expected `Result<OsString, Error>`, found `Error`
   |
   = note: expected enum `Result<OsString, std::io::Error>`
            found struct `std::io::Error`
help: try wrapping the expression in `Err`
   |
28 ~         Err(io::Error::new(
29 |             io::ErrorKind::Other,
30 |             "hostname could not be found on unsupported platform",
31 ~         ))
   |

For more information about this error, try `rustc --explain E0308`.
```
So if the goal of setting up CI is to get the same result after the PR as before, that's not going to happen.

Reply:
Oh I see. I think you just want `Err(io::Error::new(...))`, as suggested in the error message you're showing. This particular code path is probably entirely untested in CI at present, and thus the fact that it is wrong now and has always been wrong was never noticed.

My main objection was to silently using a potentially incorrect value. I'm open to doing that in the future to get hyperlinks working in WASI after we've considered the possible alternatives. But since `std::fs::canonicalize` isn't going to work on WASI right now anyway, we should just avoid changing the semantic intent of this code.

Reply:
I've done the changes for `hostname.rs`. But then, for symmetry, I think I should have a branch  `#[cfg(not(any(windows, unix)))]`  instead of `#[cfg(target_os = "wasi")]` in `hyperlink.rs`. But what should `from_path()` return in that case?

Reply:
`None`

Reply:
With a DEBUG log message. Mentioned here: https://github.com/BurntSushi/ripgrep/pull/2787#discussion_r1576137687